ui/app/src/state/reducers/uiConfig.ts

```diff
@@ -70,53 +46,62 @@ const initialState: GlobalState['uiConfig'] = {
       minute: 'numeric'
     }
   },
-  publishing: {
-    submissionCommentMaxLength: 250
-  },
-  references: null
+  references: null,
+  xml: null
 };
 
 const reducer = createReducer<GlobalState['uiConfig']>(initialState, {
-  [changeSite.type]: (state) => ({ ...initialState, globalNavigation: state.globalNavigation }),
+  [changeSite.type]: (state) => ({ ...initialState }),
   [fetchSiteUiConfig.type]: (state, { payload: { site } }) => ({
     ...state,
     isFetching: true,
     currentSite: site
   }),
-  [fetchSiteUiConfigComplete.type]: (state, { payload }) => ({
-    ...state,
-    isFetching: false,
-    ...payload
-  }),
+  [fetchSiteUiConfigComplete.type]: (state, { payload }) => {
+    let config = payload.config;
+    const references = {};
+    if (config) {
+      const configDOM = fromString(config);
+      const site = payload.site;
+      const arrays = ['widgets', 'roles', 'excludes', 'devices', 'values', 'siteCardMenuLinks', 'tools'];
+      const renameTable = { permittedRoles: 'roles' };
+
+      configDOM.querySelectorAll(':scope > references > reference').forEach((tag) => {
+        references[tag.id] = applyDeserializedXMLTransforms(deserialize(tag.innerHTML), {
+          arrays,
+          renameTable
+        });
+      });
+
+      configDOM.querySelectorAll('configuration > reference').forEach((tag) => {
+        tag.outerHTML = references[tag.id];
+      });
+
+      configDOM.querySelectorAll('plugin').forEach((tag) => {
+        const siteAttr = tag.getAttribute('site');
+        if (siteAttr === '{site}' || siteAttr === null) {
+          tag.setAttribute('site', site);
+        }
+      });
```

Comment:
This better be prior to deserializing references. Since we have the site, could be done at service layer too.